lib/formatters/stylish.js

```diff
@@ -65,7 +65,7 @@ module.exports = function(results) {
                     message.line || 0,
                     message.column || 0,
                     messageType,
-                    message.message.replace(/\.$/, ""),
+                    message.message.search(/ \.$/) >= 0 ? message.message : message.message.replace(/\.$/, ""),
```

Comment:
Nitpick: I think it would be simpler to do something like

```js
message.message.replace(/[^ ]\.$/, "")
```

rather than using a ternary expression and another regex.